tcf_website/tests/test_department.py

```diff
@@ -0,0 +1,191 @@
+# pylint: disable=no-member, line-too-long
+"""Tests for Department model."""
+
+from random import randint, uniform
+
+from django.test import TestCase
+
+from ..models import Course, CourseGrade, Review, Semester
+from .test_utils import create_new_semester, setup
+
+
+class DepartmentTestCase(TestCase):
+    """Tests for Department model."""
+
+    def setUp(self):
+        setup(self)
+        Course.objects.all().delete()
+        Review.objects.all().delete()
+        self.courses = []
+        course_numbers = set()
+        self.latest_semester = Semester().latest()
+        for _ in range(50):
+            # Create random semester using helper method
+            create_new_semester(
+                self,
+                randint(
+                    self.latest_semester.year - 10, self.latest_semester.year
+                ),
+            )
+            # Generate random course numbers
+            while True:
```

Comment:
minor nitpick: is there no better way to do this than a randomly generated check on a while True loop?

Reply:
Done